src/integrationTest/java/uk/gov/hmcts/darts/event/service/impl/SentencingRemarksAndRetentionPolicyHandlerTest.java

```diff
@@ -0,0 +1,128 @@
+package uk.gov.hmcts.darts.event.service.impl;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+import uk.gov.hmcts.darts.common.entity.CourthouseEntity;
+import uk.gov.hmcts.darts.common.entity.UserAccountEntity;
+import uk.gov.hmcts.darts.event.model.DartsEvent;
+import uk.gov.hmcts.darts.event.service.EventDispatcher;
+import uk.gov.hmcts.darts.testutils.IntegrationBase;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.darts.common.enums.SecurityRoleEnum.TRANSCRIBER;
+import static uk.gov.hmcts.darts.testutils.data.UserAccountTestData.buildUserWithRoleFor;
+import static uk.gov.hmcts.darts.transcriptions.enums.TranscriptionStatusEnum.APPROVED;
+import static uk.gov.hmcts.darts.transcriptions.enums.TranscriptionStatusEnum.REQUESTED;
+import static uk.gov.hmcts.darts.transcriptions.enums.TranscriptionUrgencyEnum.STANDARD;
+
+class SentencingRemarksAndRetentionPolicyHandlerTest extends IntegrationBase {
+
+    private static final String SOME_CASE_NUMBER = "CASE1";
+    private static final String SWANSEA_COURTHOUSE = "SWANSEA";
+    private static final String SOME_COURTROOM = "courtroom1";
+    private static final String REQUESTER_EMAIL = "test.user@example.com";
+
+
+    @Autowired
+    private EventDispatcher eventDispatcher;
+
+    @BeforeEach
+    void setUp() {
+        Jwt jwt = Jwt.withTokenValue("test")
+            .header("alg", "RS256")
+            .claim("sub", UUID.randomUUID().toString())
+            .claim("emails", List.of(REQUESTER_EMAIL))
+            .build();
+        SecurityContextHolder.getContext().setAuthentication(new JwtAuthenticationToken(jwt));
+        dartsDatabase.createTestUserAccount();
+    }
+
+    @Test
+    void createsTranscriptionWithCorrectValues() {
+        dartsDatabase.createCourthouseUnlessExists(SWANSEA_COURTHOUSE);
+        var sentencingRemarksDartsEvent = createSentencingRemarksDartsEventFor(SWANSEA_COURTHOUSE);
+
+        eventDispatcher.receive(sentencingRemarksDartsEvent);
+
+        var persistedTranscriptions = dartsDatabase.getTranscriptionRepository().findAll();
+        var persistedTranscription = persistedTranscriptions.get(0);
```

Comment:
May want to assert that there is a transcription before calling .get(0)

Reply:
done